tests/activity_api_update_test.go

```diff
@@ -0,0 +1,340 @@
+			ScheduleToCloseTimeout: durationpb.New(1 * time.Second),
+		},
+		UpdateMask: &fieldmaskpb.FieldMask{Paths: []string{"schedule_to_close_timeout"}},
+	}
+	resp, err := s.FrontendClient().UpdateActivityOptionsById(ctx, updateRequest)
+	s.NoError(err)
+	s.NotNil(resp)
+
+	// activity should fail immediately
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		assert.NoError(t, err)
+		assert.Equal(t, 0, len(description.PendingActivities))
+		assert.Equal(t, int32(1), activityCompleted.Load())
+	}, 2*time.Second, 200*time.Millisecond)
+
+	var out string
+	err = workflowRun.Get(ctx, &out)
+	var activityError *temporal.ActivityError
+	s.True(errors.As(err, &activityError))
+	s.Equal(enumspb.RETRY_STATE_NON_RETRYABLE_FAILURE, activityError.RetryState())
+	var timeoutError *temporal.TimeoutError
+	s.True(errors.As(activityError.Unwrap(), &timeoutError))
+	s.Equal(enumspb.TIMEOUT_TYPE_SCHEDULE_TO_CLOSE, timeoutError.TimeoutType())
+
+	s.Equal(int32(1), activityCompleted.Load())
+}
+
+func (s *ActivityApiUpdateClientTestSuite) TestActivityUpdateApi_ChangeScheduleToCloseAndRetry() {
+	// change both schedule to close and retry policy
+	// initial values are chosen in such a way that activity will fail due to schedule to close timeout
+	// we change schedule to close to a longer value and retry policy to a shorter value
+	// after that activity should succeed
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	var activityCompleted atomic.Int32
+	activityFunction := func() (string, error) {
+		if activityCompleted.Load() == 0 {
+			activityErr := errors.New("bad-luck-please-retry")
+			activityCompleted.Add(1)
+			return "", activityErr
+		}
+		activityCompleted.Add(1)
+		return "done!", nil
+	}
+
+	// make scheduleToClose shorter than retry interval
+	s.scheduleToCloseTimeout = 5 * time.Second
+
+	workflowFn := s.makeWorkflowFunc(activityFunction)
+
+	s.Worker().RegisterWorkflow(workflowFn)
+	s.Worker().RegisterActivity(activityFunction)
+
+	wfId := "functional-test-activity-update-api-schedule-to-close"
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:        wfId,
+		TaskQueue: s.TaskQueue(),
+	}
+
+	workflowRun, err := s.SdkClient().ExecuteWorkflow(ctx, workflowOptions, workflowFn)
+	s.NoError(err)
+
+	// wait for activity to start (and fail)
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		assert.Equal(t, int32(1), activityCompleted.Load())
+	}, 2*time.Second, 200*time.Millisecond)
+
+	// update schedule_to_close_timeout, make it longer
+	// also update retry policy interval, make it shorter
+	updateRequest := &workflowservicepb.UpdateActivityOptionsByIdRequest{
+		Namespace:  s.Namespace(),
+		WorkflowId: workflowRun.GetID(),
+		ActivityId: "activity-id",
+		ActivityOptions: &activitypb.ActivityOptions{
+			ScheduleToCloseTimeout: durationpb.New(5 * time.Second),
```

Comment:
This isn't changing anything, you already set the schedule to close timeout to 5 seconds in the beginning of this test.